components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -1961,45 +1958,45 @@ void JsonRpcService::ContinueGetERC721TokenBalance(
 void JsonRpcService::GetERC721Metadata(const std::string& contract_address,
                                        const std::string& token_id,
                                        const std::string& chain_id,
-                                       GetTokenMetadataCallback callback) {
-  JsonRpcService::GetTokenMetadata(contract_address, token_id, chain_id,
-                                   kERC721MetadataInterfaceId,
-                                   std::move(callback));
+                                       GetERC721MetadataCallback callback) {
+  nft_metadata_fetcher_->GetEthTokenMetadata(
+      contract_address, token_id, chain_id, kERC721MetadataInterfaceId,
+      std::move(callback));
 }
 
 void JsonRpcService::GetERC1155Metadata(const std::string& contract_address,
                                         const std::string& token_id,
                                         const std::string& chain_id,
-                                        GetTokenMetadataCallback callback) {
-  JsonRpcService::GetTokenMetadata(contract_address, token_id, chain_id,
-                                   kERC1155MetadataInterfaceId,
-                                   std::move(callback));
+                                        GetERC1155MetadataCallback callback) {
+  nft_metadata_fetcher_->GetEthTokenMetadata(
+      contract_address, token_id, chain_id, kERC1155MetadataInterfaceId,
+      std::move(callback));
 }
 
-void JsonRpcService::GetTokenMetadata(const std::string& contract_address,
-                                      const std::string& token_id,
-                                      const std::string& chain_id,
-                                      const std::string& interface_id,
-                                      GetTokenMetadataCallback callback) {
+void JsonRpcService::GetEthTokenUri(const std::string& chain_id,
+                                    const std::string& contract_address,
+                                    const std::string& token_id,
+                                    const std::string& interface_id,
+                                    GetEthTokenUriCallback callback) {
   auto network_url = GetNetworkURL(prefs_, chain_id, mojom::CoinType::ETH);
   if (!network_url.is_valid()) {
     std::move(callback).Run(
-        "", mojom::ProviderError::kInvalidParams,
-        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+        GURL(), mojom::ProviderError::kInternalError,
```

Comment:
I reviewed and could not find a good reason, so I have reverted this change.

I believe there were is at least one circumstance where error messages were changed in order so that it was compatible with both ProviderError and SolanaProviderError (kUnsupportedMethod -> kInternalError), but this was not that case.